### DISCUSSION TITLE
Snapstart warmup/ JIT native compilation of  QuarkusStreamHandler etc

### DISCUSSION THREAD

**Discussion by TonyWeston** [quarkusio/quarkus#35954] [Community]
Hi,
I'm looking at getting the absolute maximum response speed from Quarkus on AWS Lambda, I am getting around 4ms. I think it could be quicker.....

I notice that QuarkusStreamHandler , Quarkus.manualStart() , AmazonLambdaRecorder.handle is run every single request.  

I was considering adding 1000's of fake calls to these classes in the  crac.resource beforeCheckpoint method....  But, then I thought, this is not something I should be doing!!

So, perhaps a feature request (if its not already implemented?) - if snapstart is detected,  Quarkus should exercise these methods as part of the deployment stage, to get the JIT to compile them to native, before the snapshot is made.

Opinions?

**Comment by maxandersen** [comment]
Yeah, the idea behind the snapstart is that extension Can adapt so if you found a hot path that can be fixed reliably without too much app specific context PRs welcome to improve the snapstart/crac integration.
